File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        char_int_mpp=defaultdict(int)
        maxi=float('-inf')
        for it in s:
            char_int_mpp[it]+=1
        for it2 in char_int_mpp:
            if char_int_mpp[it2]%2!=0:
                odd=char_int_mpp[it2]
                for it1 in char_int_mpp:
                    if char_int_mpp[it1]%2==0:
                        even=char_int_mpp[it1]
                        maxi=max(maxi,odd-even)
        
        return maxi

        

File: 3753-maximum-difference-between-even-and-odd-frequency-i/3753-maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        unordered_map<int,int>mpp;
        for(auto it:s){
            mpp[it]++;
        }
        int even=INT_MAX;
        int odd =INT_MIN;
        for(auto it:mpp){
            if(it.second%2==0){
                even=min(even,it.second);
            }
            else{
                odd=max(odd,it.second);
            }
        }
        return odd-even;

        
    }
};